r/tests/testthat/test-dplyr-funcs-datetime.R

```diff
@@ -711,3 +711,48 @@ test_that("am/pm mirror lubridate", {
   )
 
 })
+
+test_that("semester works with temporal types", {
+  test_df <- tibble(
+    month_as_int = c(1:12, NA),
+    month_as_char_pad = ifelse(month_as_int < 10, paste0("0", month_as_int), month_as_int),
+    dates = as.Date(paste0("2021-", month_as_char_pad, "-15"))
+  )
+
+  # test extraction from dates
+  compare_dplyr_binding(
+     .input %>%
+      mutate(sem_wo_year = semester(dates),
+             sem_w_year = semester(dates, with_year = TRUE)) %>%
+      collect(),
+     test_df
+  )
+})
+
+test_that("semester errors with integers and characters", {
+  test_df <- tibble(
+    month_as_int = c(1:12, NA),
+    month_as_char_pad = ifelse(month_as_int < 10, paste0("0", month_as_int), month_as_int),
```

Comment:
```suggestion
    month_as_char_pad = sprintf("%02i", month_as_int),
```

This gets what you need here, yeah?

Reply:
Done.